File: src/dataset/go/armstrong_numbers_Exercism_go_rootulp/openAI/ChatGPT4_armstrong_numbers_v1.go
package armstrong

func IsNumber(n int) bool {
	return n == armstrongSum(n)
}

func armstrongSum(n int) (sum int) {
	digits := countDigits(n)
	temp := n
	for temp > 0 {
		d := temp % 10
		sum += intPow(d, digits)
		temp /= 10
	}
	return sum
}

func countDigits(n int) int {
	count := 0
	for n > 0 {
		n /= 10
		count++
	}
	return count
}

func intPow(n int, power int) int {
	result := 1
	for power > 0 {
		if power%2 == 1 {
			result *= n
		}
		n *= n
		power /= 2
	}
	return result
}

File: src/dataset/c/armstrong-numbers_exercism-Manoloon/openAI/ChatGPT4_armstrong_numbers_v1.c
#include "armstrong_numbers.h"
#include <stdint.h>
#include <stdbool.h>

bool is_armstrong_number(int candidate)
{
    if (candidate < 0) return false;
    if (candidate < 10) return true;

    int result = 0;
    int num = candidate;
    int digits = 0;
    int powers[10] = {0};

    // Calculate the number of digits and precompute powers
    for (int temp = candidate; temp > 0; temp /= 10, digits++);
    for (int i = 0; i < 10; i++) {
        int power = 1;
        for (int j = 0; j < digits; j++) {
            power *= i;
        }
        powers[i] = power;
    }

    // Calculate Armstrong sum
    while (num > 0) {
        int r = num % 10;
        result += powers[r];
        num /= 10;
    }

    return result == candidate;
}